clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/caching/agent/KubernetesCoreCachingAgent.java

```diff
@@ -0,0 +1,108 @@
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.DAEMON_SET;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.DEPLOYMENT;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.EVENT;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.HORIZONTAL_POD_AUTOSCALER;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.INGRESS;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.JOB;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.NETWORK_POLICY;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.PERSISTENT_VOLUME;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.PERSISTENT_VOLUME_CLAIM;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.POD;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.REPLICA_SET;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.ROLE;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.ROLE_BINDING;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.SECRET;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.SERVICE;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.SERVICE_ACCOUNT;
+import static com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind.STATEFUL_SET;
+
+@Slf4j
+public class KubernetesCoreCachingAgent extends KubernetesV2OnDemandCachingAgent {
+  KubernetesCoreCachingAgent(KubernetesNamedAccountCredentials<KubernetesV2Credentials> namedAccountCredentials,
+      KubernetesResourcePropertyRegistry propertyRegistry,
+      ObjectMapper objectMapper,
+      Registry registry,
+      int agentIndex,
+      int agentCount) {
+    super(namedAccountCredentials, propertyRegistry, objectMapper, registry, agentIndex, agentCount);
+  }
+
+  public Collection<AgentDataType> getProvidedDataTypes() {
+    List<AgentDataType> types = new ArrayList<>();
+    types.add(AUTHORITATIVE.forType(Keys.LogicalKind.APPLICATIONS.toString()));
+    types.add(AUTHORITATIVE.forType(Keys.LogicalKind.CLUSTERS.toString()));
+
+    types.addAll(primaryKinds().stream().map(k -> AUTHORITATIVE.forType(k.toString())).collect(Collectors.toList()));
+
+    return Collections.unmodifiableSet(new HashSet<>(types));
+  }
+
+  @Override
+  protected List<KubernetesKind> primaryKinds() {
+    return Arrays.asList(
```

Comment:
Ideally this list would be generated such that any dynamically registered `KubernetesKind` would be included here as well.